### PR TITLE
Remove some ci environment variables

### PR DESCRIPTION
**Pull request by 6543**
remove some environment variables who are either the same value or make no sense to exist in the first place:

CI_STEP_STATUS
CI_STEP_FINISHED
CI_PIPELINE_STATUS
CI_PIPELINE_FINISHED
CI_COMMIT_URL

### CODE REVIEW

**Comment by anbraten** [comment]
Could we remove those things just-in-time when we release 3.0. Those PRs will probably be quite outdated and need updates as we are constantly adding new deprecations til we release 3.0.

**Comment by anbraten** [comment]
> We should do 3.0 soon I think. There are enough breaing changes and deprecations collected. Maybe after 2.7.0?

I would like to like to do it earliest end of the year. We should provide the community some time between major releases and the breaking changes we would currently apply aren't that critical IMO.

**Comment by zc-devs** [comment]
I would add `DRONE_*` to the list.

**Comment by 6543** [comment]
> Could we remove those things just-in-time when we release 3.0. Those PRs will probably be quite outdated and need updates as we are constantly adding new deprecations til we release 3.0.

yes I just want to start adding the changes here and I'll peridically resolve conflicts and update it etc ... as it's easuly forgotten otherwise

**Comment by 6543** [comment]
@qwerty287 done

File: agent/tracer.go [hunk 1]
```diff
@@ -68,21 +68,12 @@ func (r *Runner) createTracer(ctxMeta context.Context, logger zerolog.Logger, wo
...
 		state.Pipeline.Step.Environment["CI_PIPELINE_STARTED"] = strconv.FormatInt(state.Pipeline.Started, 10)
-		state.Pipeline.Step.Environment["CI_PIPELINE_FINISHED"] = strconv.FormatInt(time.Now().Unix(), 10)
 
-		state.Pipeline.Step.Environment["CI_STEP_STATUS"] = "success"
 		state.Pipeline.Step.Environment["CI_STEP_STARTED"] = strconv.FormatInt(state.Pipeline.Started, 10)
-		state.Pipeline.Step.Environment["CI_STEP_FINISHED"] = strconv.FormatInt(time.Now().Unix(), 10)
 
 		state.Pipeline.Step.Environment["CI_SYSTEM_PLATFORM"] = runtime.GOOS + "/" + runtime.GOARCH
 
-		if state.Pipeline.Error != nil {
```
[the remark this answers is not in the document]
Comment (6543) [reply agent/tracer.go:81]: yes but that one should not indirectly be checked via environment variables ... I can a look dedicated into this maters ...

[the remark this answers is not in the document]
Comment (6543) [reply agent/tracer.go:85]: no that code is just uselless ...

as the runs_on eval for error directly is calculated in the queue impl. (task.DepMap)

[the remark this answers is not in the document]
Comment (6543) [reply agent/tracer.go:85]: -> https://github.com/woodpecker-ci/woodpecker/pull/4088

**Comment by lafriks** [comment]
Can't pipeline status be failed if step is executed even when previous steps failed so that you can notify pipeline status

**Comment by 6543** [comment]
The previous variables are not touched ans are still here ...

**Comment by 6543** [comment]
see https://github.com/woodpecker-ci/woodpecker/pull/3846/files#diff-1e99cd2f64fba341b6c56f31035dafab27d162723b979e023de5858ae1ffaa32L122

@lafriks

**Comment by 6543** [comment]
> Can't pipeline status be failed if step is executed even when previous steps failed so that you can notify pipeline status?

well the point is ... a whole pipeline status is completely irrelevant in a step ... enven within the workflow context it is hard to be usefull

and we still have some missmatch because pipeline != workflow but the vars use workflow metadata and prefix it pipeline :/

**Comment by 6543** [comment]
tldr: I dont see any any use in having it in as it is now, if we want something similar it shold be implemented with a new method and with the current terminology in mind

**Comment by xoxys** [comment]
@woodpecker-ci/maintainers Why have we removed `CI_PIPELINE_STATUS` instead of fixing it? How would notification plugins in a pipeline now know if a pipeline has succeeded/failed? Or is there no way anymore to use notification plugins to sent notifications on e.g. failed pipelines?
